adaptive_navigation/lib/scaffold.dart

```diff
@@ -52,105 +46,132 @@ class AdaptiveNavigationScaffold extends StatefulWidget {
 
 class _AdaptiveNavigationScaffoldState
     extends State<AdaptiveNavigationScaffold> {
+  NavigationType _defaultNavigationTypeResolver(BuildContext context) {
+    if (_isLargeScreen(context)) {
+      return NavigationType.drawer;
+    } else if (_isMediumScreen(context)) {
+      return NavigationType.navigationRail;
+    } else {
+      return NavigationType.bottomNavigation;
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
-    // Show a Drawer
-    if (_isLargeScreen(context)) {
-      return Row(
-        children: [
-          Drawer(
-            child: Column(
-              children: [
-                DrawerHeader(
-                  child: Center(
-                    child: widget.title,
-                  ),
+    final NavigationTypeResolver navigationTypeResolver = widget.navigationTypeResolver ?? _defaultNavigationTypeResolver;
+    switch(navigationTypeResolver(context)) {
+      case NavigationType.bottomNavigation:
+        // Show a Scaffold with a BottomNavigationBar.
+        return Scaffold(
+          body: widget.body,
+          appBar: AppBar(title: widget.title),
+          bottomNavigationBar: BottomNavigationBar(
+            items: [
+              ...widget.destinations.map(
```

Comment:
Is the for-loop preferred here instead of a map?

Reply:
Yep, done